### PR TITLE
Add isStateChangeAllowed function, update slacker2 & mixcloud

### PR DESCRIPTION
**Pull request by Paszt**
isStateChangeAllowed function allows the mutationObserver to ignore certain states, for example if an advert is playing.  Updated slacker2 & mixcloud connectors to include isStateChangeAllowed.  Added trackArtImageSelector to slacker2.

### CODE REVIEW

**Comment by inverse** [comment]
:+1: Awesome work!

File: core/content/starter.js [hunk 1]
```diff
@@ -34,7 +34,11 @@
 	if (Connector.playerSelector !== null) {
 		console.log('Web Scrobbler: Setting up observer');
 
-		var observer = new MutationObserver(Connector.onStateChanged);
+		var observer = new window.MutationObserver(function () {
```
Comment (inverse) [inline core/content/starter.js]: Do we need to add `window.`?

Comment (Paszt) [inline core/content/starter.js]: jshint checks for undefined variables.  It wouldn't pass the grunt test without including window.

**Comment by david-sabata** [comment]
Looks great, thanks! :+1:

**Pull request merged**